lookout/style/format/rules.py

```diff
@@ -201,13 +202,17 @@ def wrapped_check_fittable(self: "Rules", *args, **kwargs):
     @_check_fittable
     def fit(self, X: numpy.ndarray, y: numpy.ndarray) -> "Rules":
         """
-        Trains the rules using the base tree model and the samples (X, y). The samples may be
-        different from the ones the base model was trained on (actually, they should).
+        Trains the rules using the base tree model and the samples (X, y). If `base_model` is
+        already fitted, the samples may be different from the ones that were used.
 
         :param X: input features.
         :param y: input labels - the same length as X.
         :return: self
         """
+        try:
+            check_is_fitted(self.base_model, 'tree_')
```

Comment:
We should move this check to the constructor and always require the model to be fitted.

Reply:
I checked the fitting in the `fit` method to allow for idiomatic sklearn workflows, in particular CV. Requiring the base model to be fit in the constructor prevents from using almost all of sklearn.model_selection.